### DISCUSSION TITLE
Self-contained version

### DISCUSSION THREAD

**Discussion by Alfort-G** [niklasvh/html2canvas#2902] [General]
Hi
I am developing an app in Xamarin Forms for iOS and the only way I think I can use html2canvas would be to pass its code, like the content of html2canvas.min.js, in a string to the web browser component through method WebView.EvaluateJavaScriptAsync(String), but that way I would be missing the rest of libraries that html2canvas depends on. If this makes sense, is there a version of html2canvas that has all the pure javascript it needs?
Thank you

**Comment by coder0107git** [comment]
I think what you need is [html2canvas.min.js](https://html2canvas.hertzen.com/dist/html2canvas.min.js). It is has all of it's dependencies bundle with it (demonstrated in this [example](https://jsfiddle.net/zenjk40t/)).

**Reply by Alfort-G** [reply]
Many thanks, I was testing it copying the entire content of html2canvas.min.js to the JavaScript console in Safari or Google Chrome when I had loaded web page https://www.bbc.co.uk/news. For a reason I don't know, when I was trying to use the html2canvas function I was getting an error that it was not found. If I load a different web page like https://www.theguardian.com/uk the html2canvas function is found and can be used. I have also noticed that if I load html2canvas.min.js twice in https://www.bbc.co.uk/news then I can use the html2canvas function.
